getting_started/workflow/best_practices/what_are_godot_classes.rst

```diff
@@ -86,53 +91,43 @@ let's see a simple example of creating a single Node as a child.
 
             public Main()
             {
-                Child = new Node(); // Construct the child.
-                Child.Name = "Child"; // Change its name.
-                Child.Script = (Script)ResourceLoader.Load("child.gd"); // Give it custom features.
-                Child.Owner = this; // Serialize this node if this is saved.
-                AddChild(Child); // Add "Child" as a child of this.
+                Child = new Node();
+                Child.Name = "Child";
+                Child.Script = (Script)ResourceLoader.Load("child.gd");
```

Comment:
It would be nice to have an example relying on a `.cs` script, CC @neikeq. Would `Child.Script = (Script)ResourceLoader.Load("child.cs");` work in theory?

Reply:
Assigning a C# script will dispose the existing managed instance:

``` C#
var node = new Node();
ulong nodeId = node.GetInstanceId();

node.Script = ResourceLoader.Load<Script>("MyCustomNode.cs");

// Assert this managed instance was just disposed
Debug.Assert(node.NativeInstance == IntPtr.Zero);

// The native instance now has a new managed instance of type MyCustomNode
node = (MyCustomNode) GD.InstanceFromId(nodeId);

AddChild(node);
```

The reason this works in GDScript is because there's no such managed instance. It's just a `Variant` holding a pointer to the native instance.

If the type of the managed instance is an engine type (no custom derived class) and you're assigning a non-C# script, then it should work (this is what the example above does):

``` C#
var node = new Node();

node.Script = ResourceLoader.Load<Script>("child.gd");

// Assert this managed instance was NOT disposed
Debug.Assert(node.NativeInstance != IntPtr.Zero);

AddChild(node);
```

However, if type of the managed instance is a custom derived class, then the instance is again disposed:

``` C#
var node = new MyCustomNode();
ulong nodeId = node.GetInstanceId();

node.Script = ResourceLoader.Load<Script>("child.gd");

// Assert this managed instance was just disposed
Debug.Assert(node.NativeInstance == IntPtr.Zero);

// The native instance now has a new managed instance of whatever engine type "child.gd" inherits
node = (MyCustomNode) GD.InstanceFromId(nodeId);

// Assuming "child.gd" inherits Node2D
Debug.Assert(node.GetType() == typeof(Node2D));

AddChild(node);
```

I've yet to document this behavior, and maybe I could improve it so it's less verbose than using `InstanceId`.

Reply:
Thanks for the detailed explanation. I guess for the purpose of this tutorial explaining those details about the managed instance might be a bit out of scope, so we can probably be lazy and keep things as is under the assumption that we're in the second case (type of the managed instance is an engine type (no custom derived class) and you're assigning a non-C# script).

Once this is documented in more details somewhere else, it might be worth adding a `.. note::` mention here to link to that documentation for details; or alternatively refactor the example to avoid having to discuss this at all here.